docs: Added OpenLLMetry integration (#633)
qdrant/landing_page: qdrant-landing/content/documentation/frameworks/openllmetry.md

---
title: OpenLLMetry
weight: 2300
---

# OpenLLMetry

OpenLLMetry from [Traceloop](https://www.traceloop.com/) is a set of extensions built on top of [OpenTelemetry](https://opentelemetry.io/) that gives you complete observability over your LLM application.

OpenLLMetry supports instrumenting the `qdrant_client` Python library and exporting the traces to various observability platforms, as listed [here](https://www.traceloop.com/docs/openllmetry/integrations/introduction).

## Usage

Install the SDK

```console
pip install traceloop-sdk
```

> Instantiate the SDK

Reviewer: ```suggestion
1. Instantiate the SDK:
```

```python
from traceloop.sdk import Traceloop

> Traceloop.init()

Reviewer: I tried this, and got the following error message:

```
>>> Traceloop.init()
Error: Missing Traceloop API key, go to https://app.traceloop.com/settings/api-keys to create one
Set the TRACELOOP_API_KEY environment variable to the key
```

I tried registering, got an API key. https://app.traceloop.com/trace suggested that I replace this line with:

``` 
Traceloop.init(disable_batch=True, api_key="<my_api_key>")
```

I got about 10 lines of traceback, followed by 

`AttributeError: type object 'QdrantClient' has no attribute 'upload_points'`

Reviewer: Resolving. With qdrant-client v1.7.3, this works.

```

You're now tracing your `qdrant_client` usage with OpenLLMetry!

## Without the SDK

> Since the instrumentations are provided as standard OpenTelemetry instrumentations, you can use them as standalone packages.
> 
> Install the package.

Reviewer: ```suggestion
Since Traceloop provides standard OpenTelemetry instrumentations, you can use them as standalone packages. To do so, follow these steps: 

1. Install the package:
```

```console
pip install opentelemetry-instrumentation-qdrant
```

Instantiate the `QdrantInstrumentor`.

```python
from opentelemetry.instrumentation.qdrant import QdrantInstrumentor

> QdrantInstrumentor().instrument()

Reviewer: This didn't work for me, I got the same final error message:

`AttributeError: type object 'QdrantClient' has no attribute 'upload_points'`

Author: Can your try updating `qdrant_client` to latest? This method was added to a more recent version.

Reviewer: Ah, I was at v1.7.0. I've suggested one more change here; https://github.com/qdrant/landing_page/pull/633/files#r1500707414

```

## Further Reading

> - 📚 OpenLLMetry [API reference](https://www.traceloop.com/docs/openllmetry/introduction)

Reviewer: I think this is a better link to the OpenLLMetry API ref:

```suggestion
- 📚 OpenLLMetry [API reference](https://www.traceloop.com/docs/api-reference/introduction)
```
